Author: Federico Tello Gentile <federicotg@gmail.com>
Description: Wrap logging calls with if(){} statements to prevent
 sun.awt.X11.XEvent from creating 600 MB of char[] for no good reason
Bug-Vendor: http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=7104625
Bug-Debian: http://bugs.debian.org/651423
From: http://hg.openjdk.java.net/jdk7u/jdk7u-dev/jdk/rev/9cc780355c09

--- openjdk/jdk/src/solaris/classes/sun/awt/X11/XComponentPeer.java.orig	Fri Oct 14 03:02:15 2011 +0100
+++ openjdk/jdk/src/solaris/classes/sun/awt/X11/XComponentPeer.java	Thu Dec 08 23:28:48 2011 +0100
@@ -475,12 +475,16 @@
         if (true) {
             switch(e.getID()) {
               case PaintEvent.UPDATE:
-                  log.finer("XCP coalescePaintEvent : UPDATE : add : x = " +
+                  if (log.isLoggable(PlatformLogger.FINER)) {
+                      log.finer("XCP coalescePaintEvent : UPDATE : add : x = " +
                             r.x + ", y = " + r.y + ", width = " + r.width + ",height = " + r.height);
+                  }
                   return;
               case PaintEvent.PAINT:
-                  log.finer("XCP coalescePaintEvent : PAINT : add : x = " +
+                  if (log.isLoggable(PlatformLogger.FINER)) {
+                      log.finer("XCP coalescePaintEvent : PAINT : add : x = " +
                             r.x + ", y = " + r.y + ", width = " + r.width + ",height = " + r.height);
+                  }
                   return;
             }
         }
@@ -1257,7 +1261,9 @@
      * ButtonPress, ButtonRelease, KeyPress, KeyRelease, EnterNotify, LeaveNotify, MotionNotify
      */
     protected boolean isEventDisabled(XEvent e) {
-        enableLog.finest("Component is {1}, checking for disabled event {0}", e, (isEnabled()?"enabled":"disable"));
+        if (enableLog.isLoggable(PlatformLogger.FINEST)) {
+            enableLog.finest("Component is {1}, checking for disabled event {0}", e, (isEnabled()?"enabled":"disable"));
+        }
         if (!isEnabled()) {
             switch (e.get_type()) {
               case XConstants.ButtonPress:
@@ -1267,7 +1273,9 @@
               case XConstants.EnterNotify:
               case XConstants.LeaveNotify:
               case XConstants.MotionNotify:
-                  enableLog.finer("Event {0} is disable", e);
+                  if (enableLog.isLoggable(PlatformLogger.FINER)) {
+                      enableLog.finer("Event {0} is disable", e);
+                  }
                   return true;
             }
         }
